demos/timing_array.cc

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under both the 3-Clause BSD License and the GPLv2, found in the
+ * LICENSE and LICENSE.GPL-2.0 files, respectively, in the root directory.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause OR GPL-2.0
+ */
+
+#include "timing_array.h"
+
+#include <algorithm>
+#include <cstdint>
+#include <limits>
+#include <vector>
+
+#include "asm/measurereadlatency.h"
+#include "instr.h"
+#include "utils.h"
+
+TimingArray::TimingArray() {
+  // Explicitly initialize the elements of the array.
+  //
+  // It's not important what value we write as long as we force *something* to
+  // be written to each element. Otherwise, the backing allocation could be a
+  // range of zero-fill-on-demand (ZFOD), copy-on-write pages that all start
+  // off mapped to the same physical page of zeros. Since the cache on modern
+  // Intel CPUs is physically tagged, some elements might map to the same cache
+  // line and we wouldn't observe a timing difference between reading accessed
+  // and unaccessed elements.
+  for (int i = 0; i < size(); ++i) {
```

Comment:
I might use size_t in here but it's not a big deal.

Reply:
Yeah, `size_t` would match the API but the Google C++ style guide suggests `int` here: https://google.github.io/styleguide/cppguide.html#Integer_Types

I tried to make sure to use `size_t` on API boundaries so that it's only ever an implementation concern. Of course, here `i` will really only ever range from 0 to 255 so it's not too worrying.